src/python/zquantum/qaoa/farhi_ansatz_test.py

```diff
@@ -103,5 +118,17 @@ def test_generate_circuit(self):
         evaluated_circuit = parametrized_circuit.evaluate(self.symbols_map)
         final_unitary = evaluated_circuit.to_unitary()
 
+        # Then
+        self.assertTrue(compare_unitary(final_unitary, self.target_unitary, tol=1e-10))
+
+    def test_generate_circuit_with_ising_operator(self):
+        # When
+        self.cost_hamiltonian = change_operator_type(
+            self.cost_hamiltonian, IsingOperator
+        )
+        parametrized_circuit = self.ansatz._generate_circuit()
+        evaluated_circuit = parametrized_circuit.evaluate(self.symbols_map)
+        final_unitary = evaluated_circuit.to_unitary()
```

Comment:
I don't think this is actually behaving as intended. You never reset `ansatz.cost_hamiltonian` so the cost hamiltonian used to generate the circuit is still the old one.

Reply:
Right, thanks for spotting this! Fixed here: https://github.com/zapatacomputing/z-quantum-qaoa/pull/27/commits/84e71b250b45ed2030d7465eb3f1cf8add866d24